bip-XXXX-Bech32_Encoded_Transaction_Position_References.mediawiki

```diff
@@ -0,0 +1,278 @@
+</ref>
+
+==== Namespacing ====
+* Every non-compatible system MUST use a different '''magic code'''.
+* The magic code MUST be prefixed to the data payload, and included in the Bech32 encoding.
+* Applications MUST reject codes with unknown magic codes.
+* Applications MUST display an error for known, but unsupported magic codes.
+
+=== Magic Codes ===
+* Magic codes are either 5, 10, 15, or 20-bits long.
+
+* The length of the magic code is set by the first 5 bits:
+** 0x0 to 0x7: 5-bit long. (All Reserved<ref>'''Why reserve all single character Magic Codes?''' The authors of this specification see many potential Bitcoin Related uses for additional magic codes, such as with Sidechains and other special user facing projects.</ref> for Bitcoin-Related Use, we have the special exception for Litecoin, that is also assigned a 5-bit code.)
+** 0x8 to 0x15: 10-bit long.
+** 0x16 to 0x23: 15-bit long.
+** 0x24 to 0x31: 20-bit long.
+
+=== Compatibly ===
+There are no known compatibility issues.
+
+== Rationale ==
+
+<references />
+
+== Reference implementations ==
+C Reference Implementation: https://github.com/jonasschnelli/bitcoin_txref_code
+
+== Appendices ==
+
+=== Test Vectors ===
+This appendix contains test vectors for TxRef Standard Application Display Formats.
+
+''Please Note: All test vectors are shown to help test if a Standard Display Client is compliant or not. All real-life applications should comply with the Display Format's own Test Vectors''
+
+The following strings have a valid Human Readable Part and Bech32 Checksum.
+* <tt>TX1A12UEL5L</tt>
+* <tt>tx1an83characterlonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio1tt5tgs</tt>
+* <tt>tx1abcdef1qpzry9x8gf2tvdw0s3jn54khce6mua7lmqqqxw</tt>
+* <tt>tx11qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqc8247j</tt>
+
+The following list gives invalid TxRef's and the reason for their invalidity.
+* <tt>bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kg3g4ty</tt>: Invalid human-readable part
+* <tt>tx1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t5</tt>: Invalid checksum
+
+=== Allocated Magic Values ===
+
+This appendix contains a table of the allocated or reserved magic codes.
+* If you wish to claim a magic code, please open a pull request adding the code to this list.
+* If this list becomes too popular, we will move it to another location.
+
+==== Reserved Values: ====
+Magic Values 5-bit long:
+{| class="wikitable"
+|'''Magic Value'''
+|'''Starting Value(s)'''
+|'''Display'''
+|'''Description'''
+|-
+|
+|0x0
+|‘q’
+|Reserved
+|-
+|
+|0x1
+|‘p’
+|Reserved
+|-
+|
+|0x2
+|‘z’
+|Reserved
+|-
+|'''Bitcoin'''
+|'''0x3'''
+|'''‘r’'''
+|'''Bitcoin Main Chain'''
+|-
+|'''Litecoin'''
```

Comment:
It is not about identifying a coin, but a chain: some chains could have several "coins".
For altcoins that didn't create their own unique genesis block, a later block when it starts to diverge can be used. The point is to have less prune to collisions than a memorable string, which we would need to register somewhere.
It is sane to assume that altchains will respect BIP173's "bc" and "bt" but not so much to assume they will know about each other or will wait for some centralized registry to assign them a memorable and "unique" string.
Of course if someone wants to "collide" one can always still use bitcoin's genesis block for their altchain and "bc" to "identify it". My point is that altchains that don't want to collide with each other or with btc or any testnet still have it very easy if we rely only on the memorable string (unless we assume the centralized registry is actually maintained or we also use some chain id like the genesis block hash or the first unique block or whatever).